src/ArrayUtilities.hpp

```diff
@@ -300,92 +300,47 @@ static void stringToArray( LvArray::Array< T, NDIM, PERMUTATION, INDEX_TYPE, DAT
 }
 
-
 /**
- * @struct arrayToStringHelper
- * Helper struct/functor to output an array into a string.
+ * @brief This function outputs the contents of an array slice to an output stream.
+ * @param stream the output stream for which to apply operator<<.
+ * @param slice the slice to output.
+ * @return a reference to the ostream.
  */
-template< typename T, typename INDEX_TYPE >
-struct arrayToStringHelper
+template< typename T, int NDIM, int UNIT_STRIDE_DIM, typename INDEX_TYPE >
+std::ostream & operator<<( std::ostream & stream,
+                           LvArray::ArraySlice< T const, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE > const & slice )
 {
+  stream << "{ ";
 
-  /**
-   * @brief recursive function to loop over the contents of an array and output in a
-   *        format similar to the initializer of a standard c-array.
-   * @param[in] data pointer to the data
-   * @param[in] dims pointer to the dims array
-   * @param[in] strides pointer to the strides array
-   * @param[in/out] output the output string
-   */
-  template< int NDIM >
-  static
-  typename std::enable_if< NDIM==0, void >::type
-  dimExpansion( T const * const data,
-                INDEX_TYPE const * const dims,
-                INDEX_TYPE const * const LVARRAY_UNUSED_ARG( strides ),
-                std::string & output )
-  {
-    for( INDEX_TYPE i=0 ; i<dims[0] ; ++i )
-    {
-      output += std::to_string( data[i] );
-      if( i < dims[0] - 1 )
-      {
-        output += ", ";
-      }
-    }
-  }
+  if( slice.size( 0 ) > 0 )
+    stream << slice[ 0 ];
 
-  template< int NDIM >
-  static
-  typename std::enable_if< NDIM!=0, void >::type
-  dimExpansion( T const * const data,
-                INDEX_TYPE const * const dims,
-                INDEX_TYPE const * const strides,
-                std::string & output )
+  for( INDEX_TYPE i = 1 ; i < slice.size( 0 ) ; ++i )
   {
-    for( INDEX_TYPE i=0 ; i<dims[0] ; ++i )
-    {
-      output += "{ ";
-      dimExpansion< NDIM-1 >( &(data[ i*strides[0] ] ), dims+1, strides+1, output );
-      output += " }";
-      if( i < dims[0] - 1 )
-      {
-        output += ", ";
-      }
-
-    }
+    stream << ", " << slice[ i ];
   }
 
-};
-
+  stream << " }";
+  return stream;
+}
 
 /**
- * @brief This function converts an array to a string
- * @param array The array to convert
- * @return a string containing the contents of @p array
+ * @brief This function outputs the contents of an array view to an output stream.
+ * @param stream the output stream for which to apply operator<<.
+ * @param view the view to output.
+ * @return a reference to the ostream.
  */
 template< typename T,
           int NDIM,
           int UNIT_STRIDE_DIM,
           typename INDEX_TYPE,
           template< typename > class DATA_VECTOR_TYPE >
-static std::string arrayToString( LvArray::ArrayView< T, NDIM, UNIT_STRIDE_DIM, INDEX_TYPE, DATA_VECTOR_TYPE > const & array )
+std::ostream & operator<<( std::ostream & stream,
```

Comment:
so are the stream operators replacing the `arrayToString, stringToArray` functions in the code?...and do we need error checking for this operation? I know I did it somewhere...maybe in the xml reader. but it should be here.

Reply:
Not `stringToArray`, just `arrayToString` which shouldn't need any error checking.